### PR TITLE
ROX-9718: Make admin secret clickable

### PR DESCRIPTION
**Pull request by kurlov**
## Description

This change will bring user directly to the secret in OpenShift UI

Jira ticket: [ROX-9606](https://issues.redhat.com/browse/ROX-9718)

## Checklist
- [ ] Investigated and inspected CI test results
- [x] Unit test and regression tests added
~~- [ ] Evaluated and added CHANGELOG entry if required~~
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [stackrox/openshift-docs](https://github.com/stackrox/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

Extend an existing unit tests accordingly

## Testing Performed
Manually deployed operator. Here is how it looks with admin secret reference if it was auto-generated:
default_generated

Already existed secret
existing_password

### CODE REVIEW

File: operator/pkg/central/extensions/reconcile_admin_password.go
Comment (porridge) [inline operator/pkg/central/extensions/reconcile_admin_password.go:127]: Shouldn't this code also influence the return value like in the `if` block right above?

File: operator/pkg/central/extensions/reconcile_admin_password.go
Comment (porridge) [inline operator/pkg/central/extensions/reconcile_admin_password.go:54]: This name seems quite vague, but this field is set only in a very specific case, this seems fishy... 🤔

File: operator/pkg/central/extensions/reconcile_admin_password.go
Comment (porridge) [inline operator/pkg/central/extensions/reconcile_admin_password.go:197]: `click the secret reference below` is very console-centric and won't make sense to someone viewing the CR with something like `kubectl get centrals`. How about this instead:
```suggestion
			"To view the password see the secret reference field or run\n"+
```

File: operator/apis/platform/v1alpha1/central_types.go
Comment (porridge) [inline operator/apis/platform/v1alpha1/central_types.go:334]: I think this should be an optional field (i.e. a string pointer) since admin password can be disabled, in which case there will not be a secret to refer to here.

File: operator/pkg/central/extensions/reconcile_admin_password.go
Comment (porridge) [inline operator/pkg/central/extensions/reconcile_admin_password.go:126]: I'm somewhat worried that the association between the condition and the actions it guards is not very obvious. It *might* be correct in all cases but it seems fragile.
How about in `readPasswordFromReferencedSecret` we instead set `r.passwordSecretName`  to whatever the right secret is (or empty string if none) even if it's the default secret. Then here we would just set `...SecretReference` to either `nil` or that value. Perhaps that would remove the need for the `hasCredentials` field?
Or am I missing some corner case?

File: operator/apis/platform/v1alpha1/central_types.go
Comment (kurlov) [reply operator/apis/platform/v1alpha1/central_types.go:333]: Yes, it was a stale screenshot. I've updated it

File: operator/apis/platform/v1alpha1/central_types.go
Comment (kurlov) [reply operator/apis/platform/v1alpha1/central_types.go]: I wanted achieve the same with `omitempty` but it didn't work. Unfortunately, transforming it to string pointer did not work neither. Here is a screenshot how it looks like with the nil pointer:

disabled_password

I also looked for [CSV marker](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md) but have not found any suitable for this case. I guess reference will be `None` sometimes which is not cool 😞 
I returned reference back to be just a regular `string`

File: operator/pkg/central/extensions/reconcile_admin_password.go
Comment (kurlov) [reply operator/pkg/central/extensions/reconcile_admin_password.go]: I really like this idea. Although there is no enough information when `readPasswordFromReferencedSecret` is executed so `passwordSecretName` is changed in downstream functions. The alternative might be to rework password generating/update/reading logic so `readPasswordFromReferencedSecret` will have enough information but I am not sure about implementing such refactoring.
Additional bool var is dropped

File: operator/pkg/central/extensions/reconcile_admin_password.go
Comment (kurlov) [reply operator/pkg/central/extensions/reconcile_admin_password.go]: It is dropped thanks to your suggestion 👍

File: operator/pkg/central/extensions/reconcile_admin_password.go
Comment (kurlov) [reply operator/pkg/central/extensions/reconcile_admin_password.go]: I added `SecretReference` check so it effects which bool to return as well

**Review by SimonBaeumer — APPROVED** [review]
LGTM! Please wait for @porridge approval as he was more involved in this PR 👊

File: operator/pkg/central/extensions/reconcile_admin_password_test.go
Comment (porridge) [inline operator/pkg/central/extensions/reconcile_admin_password_test.go:113]: I've been starting at this PR for the last 5 minutes and I cannot figure out how this test case can possibly be passing?
I can see the only place where `.SecretReference` is being is set is from `r.passwordSecretName` [here](https://github.com/stackrox/stackrox/pull/1206/files#diff-0fdb3bf9ba7d7186b84a925524c29d41640249c259061d3a7dc9f6e262c631a1R122) and that field in turn is only ever set to the `defaultPasswordSecretName` constant on lines 152, 155 and 182.
Can you please explain that to me?

File: operator/apis/platform/v1alpha1/central_types.go
Comment (porridge) [reply operator/apis/platform/v1alpha1/central_types.go:334]: The screenshot above looks good to me. I guess it would be better if the UI hid the field altogether, but it's not something we have influence... Also how does it look in the no-password case when it's a non-pointer string? Let's chat about this.

**Review by porridge — COMMENTED** [review]
The code looks much simpler now, thanks. But I have some questions.

File: operator/apis/platform/v1alpha1/central_types.go
Comment (kurlov) [reply operator/apis/platform/v1alpha1/central_types.go]: `SecretReference` is set back to being a pointer

File: operator/pkg/central/extensions/reconcile_admin_password.go
Comment (porridge) [inline operator/pkg/central/extensions/reconcile_admin_password.go:122]: This worries me a little. I'm somewhat afraid that (given how this whole codebase, and **especially** status updates is made of closures upon closures) some refactoring elsewhere might cause `r.passwordSecretName` to get changed beneath our feet **before** `status` is serialized. 

How about:

```suggestion
        secretNameCopy := r.passwordSecretName
	status.Central.AdminPassword.SecretReference = &secretNameCopy
```

**Review by porridge — APPROVED** [review]
Just one thing, feel free to merge if you agree and it works.

**Pull request merged**
